test/blackbox/pushpull_authn.bats

```diff
@@ -0,0 +1,145 @@
+load helpers_cloud
+
+function setup() {
+    # Verify prerequisites are available
+    if ! verify_prerequisites; then
+        exit 1
+    fi
+
+    # Setup zot server
+    local zot_root_dir=${BATS_FILE_TMPDIR}/zot
+    local zot_config_file=${BATS_FILE_TMPDIR}/zot_config.json
+    local zot_htpasswd_file=${BATS_FILE_TMPDIR}/zot_htpasswd
+    htpasswd -Bbn test test123 >> ${zot_htpasswd_file}
+    
+    echo ${zot_root_dir} >&3
+
+    mkdir -p ${zot_root_dir}
+
+    cat > ${zot_config_file}<<EOF
+{
+  "distSpecVersion":"1.1.0-dev",
+  "storage":{
+    "dedupe": true,
+    "gc": true,
+    "gcDelay": "1h",
+    "gcInterval": "6h",
+    "rootDirectory": "${zot_root_dir}"
+  },
+  "http": {
+		"address": "127.0.0.1",
+		"port": "8080",
+    "realm":"zot",
+    "auth": {
+      "htpasswd": {
+        "path": ${zot_htpasswd_file}
+      },
+      "failDelay": 5
+    },
+    "accessControl": {
+      "repositories": {
+        "**": {
+          "anonymousPolicy": ["read"],
+          "defaultPolicy": ["read", "create"]
+        }
+      },
+      "adminPolicy": {
+        "users": ["admin"],
+        "actions": ["read", "create", "update", "delete"]
+      }
+    }
+  },
+  "log":{
+    "level":"debug",
+  }
+}
+EOF
+    zot_serve ${zot_config_file}
+    wait_zot_reachable "http://127.0.0.1:8080/v2/_catalog"
+}
+
+@test "push image with regclient" {
+    run regctl registry set localhost:8080 --tls disabled
+    run regctl registry login localhost:8080 -u test -p test123
```

Comment:
^ this is the big change, making sure **with** client auth enabled, things don't break

Reply:
Understood.